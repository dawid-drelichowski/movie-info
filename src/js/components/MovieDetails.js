import React, {Component} from 'react'
import PropTypes from 'prop-types'

export default class MovieDetails extends Component {
  static propTypes = {
    config: PropTypes.object.isRequired,
    details: PropTypes.object.isRequired
  }
  render () {
    const details = this.props.details
    return <article className="py-3">
      {details.poster_path ? this.renderImage() : ''}
      <dl>
        <dd className="float-left mb-0 mr-2">Rating:</dd>
        <dt>{details.vote_average} ({details.vote_count} votes)</dt>
        <dd className="float-left mb-0 mr-2">Genres:</dd>
        <dt>{details.genres.map(genre => genre.name).join(', ')}</dt>
        <dd className="float-left mb-0 mr-2">Release date:</dd>
        <dt>{details.release_date}</dt>
      </dl>
      <p className="text-justify m-0">{details.overview}</p>
    </article>
  }
  renderImage () {
    const config = this.props.config.movieDb.images
    return <img src={`${config.url}${config.size.big}${this.props.details.poster_path}`} className="float-left pr-2"/>
  }
}
