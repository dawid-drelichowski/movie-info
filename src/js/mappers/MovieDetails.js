import Abstract from 'mappers/Abstract'
import Details from 'models/MovieDetails'
import GenreMapper from 'mappers/MovieGenre'

export default class extends Abstract {
  static mapping = new Map([
    ['id', 'id'],
    ['title', 'title'],
    ['imagePath', 'poster_path'],
    ['averageVote', 'vote_average'],
    ['voteCount', 'vote_count'],
    ['releaseDate', 'release_date'],
    ['overview', 'overview']
  ])
  static map (data) {
    const model = new Details()
    if (data.genres.length) {
      model.genres = GenreMapper.mapCollection(data.genres)
    }
    this.performMapping(data, model)
    return model
  }
}
