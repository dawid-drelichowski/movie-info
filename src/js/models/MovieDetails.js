export default class {
  id = 0
  title = ''
  imagePath = ''
  averageVote = 0.0
  voteCount = 0
  genres = []
  releaseDate = ''
  overview = 0
}
