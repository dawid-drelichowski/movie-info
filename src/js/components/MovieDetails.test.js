import React from 'react'
import MovieDetails from 'components/MovieDetails'
import renderer from 'react-test-renderer'

describe('Movie details component', () => {
  const sharedData = {
    vote_average: 6.4,
    vote_count: 4976,
    genres: [
      {
        id: 10751,
        name: 'Family'
      },
      {
        id: 16,
        name: 'Animation'
      }
    ],
    release_date: '2015-06-17',
    overview: 'Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill...'
  }

  function getRenderedComponent (data) {
    return renderer.create(
      <MovieDetails data={data} />
    )
  }

  it('shouldn\'t have image', () => {
    expect(getRenderedComponent(sharedData).toJSON()).toMatchSnapshot()
  })

  it('should have image', () => {
    expect(getRenderedComponent({...sharedData, poster_path: '/q0R4crx2SehcEEQEkYObktdeFy.jpg'}).toJSON()).toMatchSnapshot()
  })
})
