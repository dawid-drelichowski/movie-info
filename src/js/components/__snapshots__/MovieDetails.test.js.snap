// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie details component should have image 1`] = `
<article>
  <dl>
    <dd>
      Rating:
    </dd>
    <dt>
      6.4
       (
      4976
       votes)
    </dt>
    <dd>
      Genres:
    </dd>
    <dt>
      Family,Animation
    </dt>
    <dd>
      Release date:
    </dd>
    <dt>
      2015-06-17
    </dt>
  </dl>
  <img
    src="https://image.tmdb.org/t/p/w154/q0R4crx2SehcEEQEkYObktdeFy.jpg"
  />
  <p>
    Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill...
  </p>
</article>
`;

exports[`Movie details component shouldn't have image 1`] = `
<article>
  <dl>
    <dd>
      Rating:
    </dd>
    <dt>
      6.4
       (
      4976
       votes)
    </dt>
    <dd>
      Genres:
    </dd>
    <dt>
      Family,Animation
    </dt>
    <dd>
      Release date:
    </dd>
    <dt>
      2015-06-17
    </dt>
  </dl>
  
  <p>
    Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill...
  </p>
</article>
`;
